src/NITTA/Synthesis/Allocation.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE NoMonomorphismRestriction #-}
+
+{-# OPTIONS -fno-warn-orphans #-}
+
+module NITTA.Synthesis.Allocation (
+    AllocationMetrics (..),
+) where
+
+import Data.Aeson (ToJSON)
+import qualified Data.Map as M
+import qualified Data.Text as T
+import GHC.Generics
+import NITTA.Model.Networks.Bus
+import NITTA.Model.Networks.Types
+import NITTA.Model.Problems.Allocation
+import NITTA.Model.ProcessorUnits.Types
+import NITTA.Model.TargetSystem
+import NITTA.Synthesis.Types
+import NITTA.Utils
+
+data AllocationMetrics = AllocationMetrics
+    { mParallelism :: T.Text
```

Comment:
It will be better to use ADT type, instead of string.

Reply:
I thought that we should use strict types here because metrics converts to TS types